students/idcrickmore/lesson06/test_mailroom_pt4.py

```diff
@@ -0,0 +1,39 @@
+from mailroom_pt4 import report, letters, donor_list
+import os
+
+#test mailroom functions using pytest
+
+
+def test_report():
+    actual = report()
+    expected = (
+        "-------------------- REPORT --------------------\n"
+        "Donor Name         | Total Given    | Num Gifts| Average Gift\n"
+        "---------------------------------------------------------------\n"
+        "Christiaan Huygens   $      9174.00           2  $   4587.00\n"
+        "Galileo Galilei      $      8848.00           3  $   2949.33\n"
+        "Edwin Hubble         $      1899.00           1  $   1899.00\n"
+        "Edmond Halley        $      1856.00           3  $    618.67\n"
+        "Giovanni Cassini     $       209.00           1  $    209.00\n")
+
+        
+    assert expected == actual
+    
+    
+def test_letters():
+    assert os.path.isfile('Giovanni Cassini_2018-04-30.txt')
+    assert os.path.isfile('Edmond Halley_2018-04-30.txt')
+    assert os.path.isfile('Edwin Hubble_2018-04-30.txt')
+    assert os.path.isfile('Galileo Galilei_2018-04-30.txt')
+    assert os.path.isfile('Christiaan Huygens_2018-04-30.txt')
```

Comment:
if I run this test on my machine it will fail because those files do not exist.

you are missing a call to your function to generate those files -> `mailroom_pt4.letters()` and then you can do asserts

Reply:
you can also simplify your test:
```
expected_files = ['file1', 'file2']
for ef in expected_files:
    assert os.path.isfile(ef)
```